release/Magarena/scripts/Ajani__Valiant_Protector.groovy

```diff
@@ -6,23 +6,29 @@
                 source,
                 this,
                 "PN reveals cards from the top of PN's library until PN reveals a creature card. " +
-                "Put that card into PN's hand and the rest on the bottom of PN's library in a random order."
+                "PN puts that card into PN's hand and the rest on the bottom of PN's library in a random order."
             );
         }
         @Override
         public void executeEvent(final MagicGame game, final MagicEvent event) {
             final MagicCardList library = event.getPlayer().getLibrary();
             def predicate = { final MagicCard card -> card.hasType(MagicType.Creature) };
-            final MagicCardList nonTarget = new MagicCardList(library.takeWhile({ !predicate(it) }));
-            if (library.any(predicate)) {
-                final MagicCard target = library.find(predicate);
-                game.doAction(new RevealAction(nonTarget.plus(target)));
-                game.doAction(new ShiftCardAction(target, MagicLocationType.OwnersLibrary, MagicLocationType.OwnersHand));
-            } else {
-                game.doAction(new RevealAction(nonTarget));
+            final MagicCardList revealed = new MagicCardList();
+            MagicCard target = MagicCard.NONE;
```

Comment:
mutable variable target is not necessary, use `break` to exit the while loop when top card matching predicate is found.